### PR TITLE
remove --cleanup-ipvs flag of kube-proxy

### LINKED ISSUE #96034: kube-proxy --cleanup don‘t clean ipvs rules
**What happened**:
kube-proxy don’t clean ipvs rules with `--cleanup `. 

**What you expected to happen**:
kube-proxy clean ipvs rules with `--cleanup ` by default. 

**How to reproduce it (as minimally and precisely as possible)**:
Run kube-proxy in ipvs mode, and clean rules with `kube-proxy --cleanup`.

**Anything else we need to know?**:
 Based on https://github.com/kubernetes/kubernetes/pull/76109, `--cleanup` should cleanup all rules by default, including ipvs rules. And `--cleanup-ipvs` also retains functionality.

**Environment**:
- Kubernetes version (use `kubectl version`):
v1.20.0-bate.0
- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`):
centos7
- Kernel (e.g. `uname -a`):
- Install tools:
- Network plugin and version (if this is a network-related bug):
- Others:

### PR DESCRIPTION
**Pull request by maaoBit**
Signed-off-by: maao <maao420691301@gmail.com>

**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #96034

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
kube-proxy: remove deprecated --cleanup-ipvs flag of kube-proxy, and make --cleanup flag always to flush IPVS
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: pkg/proxy/ipvs/proxier.go [hunk 1]
```diff
@@ -803,22 +803,20 @@ func cleanupIptablesLeftovers(ipt utiliptables.Interface) (encounteredError bool
...
-			klog.Errorf("Error flushing IPVS rules: %v", err)
-			encounteredError = true
-		}
+func CleanupLeftovers(ipvs utilipvs.Interface, ipt utiliptables.Interface, ipset utilipset.Interface) (encounteredError bool) {
+	// Return immediately when ipvs interface is nil - Probably initialization failed in somewhere.
+	if ipvs == nil {
+		return true
+	}
+	encounteredError = false
+	err := ipvs.Flush()
```
Comment (andrewsykim) [inline pkg/proxy/ipvs/proxier.go:812]: This is probably fine but just confirming, this flushes all IPVS rules, not just the Kubernetes managed ones right?

File: pkg/proxy/ipvs/proxier.go [hunk 2]
```diff
@@ -803,22 +803,20 @@ func cleanupIptablesLeftovers(ipt utiliptables.Interface) (encounteredError bool
...
-		encounteredError = false
-		err := ipvs.Flush()
-		if err != nil {
-			klog.Errorf("Error flushing IPVS rules: %v", err)
-			encounteredError = true
-		}
+func CleanupLeftovers(ipvs utilipvs.Interface, ipt utiliptables.Interface, ipset utilipset.Interface) (encounteredError bool) {
+	// Return immediately when ipvs interface is nil - Probably initialization failed in somewhere.
+	if ipvs == nil {
+		return true
```
Comment (andrewsykim) [inline pkg/proxy/ipvs/proxier.go:809]: if IPVS is nil we probably want to still run the iptables clean up, so maybe this should be:
```
if ipvs != nil {
    // IPVS clean up
}

// rest of the function..
```

File: pkg/proxy/ipvs/proxier.go (hunk 1 above)
Comment (maaoBit) [reply pkg/proxy/ipvs/proxier.go]: yes, this clears all ipvs rules

**Review by dougsland — COMMENTED** [review]
LGTM

File: pkg/proxy/ipvs/proxier.go [hunk 3]
```diff
@@ -803,19 +803,16 @@ func cleanupIptablesLeftovers(ipt utiliptables.Interface) (encounteredError bool
...
+func CleanupLeftovers(ipvs utilipvs.Interface, ipt utiliptables.Interface, ipset utilipset.Interface) (encounteredError bool) {
+	// Clear all ipvs rules
+	if ipvs != nil {
 		err := ipvs.Flush()
 		if err != nil {
 			klog.Errorf("Error flushing IPVS rules: %v", err)
 			encounteredError = true
 		}
 	}
+
```
[the remark this answers is not in the document]
Comment (maaoBit) [reply pkg/proxy/ipvs/proxier.go]: Has been modified, PTAL

**Review by dougsland — COMMENTED** [review]
/lgtm

**Review by andrewsykim — COMMENTED** [review]
/approve
/lgtm

Thanks @maaoBit

**Pull request merged**